docs/errors-and-warnings.md

```diff
@@ -0,0 +1,35 @@
+# Errors and Warnings
+
+These are the errors and warnings grouped by category to help you understand their purpose.
+
+> ❌ This is an error  
+> ⚠️ This is a warning  
+> 💡 Guide to help you troubleshoot  
+
+## General Errors
+
+### circular-reference
+
+❌ A series of document references creates a closed loop.
```

Comment:
Does this mean reference including tokens? This can be warning, as it like a document error with limited influence.